tests/test_search.py

```diff
@@ -136,6 +138,536 @@ def test_mongo_connect_store_file():
 
     assert mongo.grid.get(id1).read() == "hello world"
 
+def get_dummy_task(id):
+    t = Task(id=id)
+    t.target = ""
+    t.category = ""
+    t.timeout = ""
+    t.package = ""
+    t.options = ""
+    t.priority = 0
+    t.custom = ""
+    t.owner = ""
+    t.machine = ""
+    t.platform = ""
+    t.submit_id = id
+    return t
+
+def do_search(report1, report2, term, value):
+    ## create temp CWD
+    set_cwd(tempfile.mkdtemp())
+    cuckoo_create(cfg={
+        "reporting": {
+            "mongodb": {
+                "enabled": True,
+                "db": "cuckootest",
+            },
+        },
+    })
+
+    ## Setup MongoDB
+    mongo.init()
+    assert mongo.database == "cuckootest"
+    mongo.drop()
+    mongo.connect()
+
+    ## Setup Database
+    db = Database()
+    db.connect()
+    session = db.Session()
+
+    ## Inserting 1st report
+    t = get_dummy_task(report1['info']['id'])
+    session.add(t)
+    report = report1
+    mongo.db.analysis.save(report)
+    ## Inserting 2nd report
+    t = get_dummy_task(report2['info']['id'])
+    session.add(t)
+    report = report2
+    mongo.db.analysis.save(report)
+
+    ## Commit the changes
+    session.commit()
+
+    result = searcher.find(term, value)
+
+    assert len(result) == 1
+    assert result[0]['id'] == report1['info']['id']
+
+def test_mongo_file_name():
+    report1 = {'target':
+                  {'file':
+                       {'name': 'test1'}
+                   },
+              'info':
+                  {'id':1000,
+                   'category':'url'}
+              }
+
+    report2 = {'target':
+                  {'file':
+                       {'name': 'test2'}
+                   },
+              'info':
+                  {'id': 1001,
+                   'category':'url'}
+              }
+
+    do_search(report1, report2, "name", report1['target']['file']['name'])
+    do_search(report2, report1, "name", report2['target']['file']['name'])
```

Comment:
I'm sure the code quality of these tests could be improved ;-)
E.g., each of these file thingies could be abstracted away and searched through with one or two unittests in much less code.